### DISCUSSION TITLE
IIS sometimes returns a empty static file

### DISCUSSION THREAD

**Discussion by alonstar** [dotnet/aspnetcore#28408] [Q&A]
hi everyone,

I got a very strange question in my AspNetCore 3.1 site.

IIS sometimes send a empty css (or js ) file with 200 status code, If I restart iis can fix this issue.

When it happens, I can find http request in IIS Log but not log in Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.
(Usually It logs "The file "/js/main.build.js" was not modified" OR "Sending file".)

I didn't find any errors, and I tried to set a new website or update all files...etc but still can't resolve this question.

Please tell me how to find out what's wrong between IIS and AspNetCore

**Comment by lextm** [comment]
To trace the source code empty 200 responses, your most convenient option is [Failed Request Tracing rules on IIS](https://learn.microsoft.com/en-us/troubleshoot/developer/webapps/iis/health-diagnostic-performance/troubleshoot-failed-requests-using-tracing-in-iis-85).
